Test if popup is open before opening new popup

ryanjhodge opened the pull request:
Fixes issue related to #28 in which popups were out of order in respect to layer z index. Specifically when geojson vectors were drawn inside cartodb polygon (example provided by @ryanbranciforte). 

Closes #28.  Let's keep trailheadlabs/outerspatial#1267 for builder UI overlay ordering enhancement discussion.

ryanbranciforte commented:
@ryanjhodge Thanks for working on this! Still having the popup issue in this map builder map - check it out here: https://www.outerspatial.com/builder_maps/2/embed#11.5/41.3930/-70.6359

Basically, when you click on the trail, the polygon popup covers up the trail popup.

nateirwin commented:
This hasn't been merged yet, so you won't see it until that's done.

ryanbranciforte commented:
Got it. @nateirwin Just post a note that it's been merged and I can test after.

nateirwin commented:
Hey, @ryanjhodge: One big question about this: Will this change work effectively with other overlays? So, for example, if a popup associated with a marker is displaying, what will happen if I click on a CARTO geometry elsewhere on the map?

ryanjhodge commented:
Looks like it works in that case. I added point, line, and polygon geojson layers overlapping a carto polygon layer. Clicking between layers causes the open popup to close.  The geojson overlay popups take precedence over the carto popups when they overlap.

I also added overlapping carto line and carto polygon layers, which are handled using a nested popup.




In the first case  and in the second both layers are handled and merge

nateirwin commented on src/map.js:
```diff
@@ -476,15 +476,17 @@ MapExt = L.Map.extend({
               }
 
               if (actual.length) {
```
You can do all the checks in a single line, and just checking for `!me._popup` should do the trick  as well. So line 478 would change to:

```
if (actual.length && !me._popup) {
```

ryanjhodge commented:
Good call.  Made that change and tested it.

ryanbranciforte commented:
Not sure if this issue was meant to resolve the issue on the SMF map, but still seeing issue in this Builder Map.

https://www.outerspatial.com/builder_maps/2/embed#16.5/41.42183/-70.69521

It brings up the Area info box instead of the Trail info box.
![screen shot 2017-06-23 at 9 45 22 am](https://user-images.githubusercontent.com/2602563/27491843-ce03c282-57f8-11e7-8364-c810a6786bf0.png)

nateirwin commented:
We have to deploy a new version of the JS library before it will show up in the map.

ryanbranciforte commented:
Got it.
